drivers/kscan/kscan_ft5336.c

```diff
@@ -204,6 +245,13 @@ static int ft5336_init(const struct device *dev)
 
 	k_work_init(&data->work, ft5336_work_handler);
 
+#ifdef CONFIG_KSCAN_FT5336_INTERRUPT_POLLING_MODE
+	if (i2c_reg_write_byte(data->i2c, config->i2c_address, REG_G_MODE,
+			       REG_G_MODE_INTERRUPT_POLLING) < 0) {
+		LOG_ERR("Could not configure interrupt polling mode.");
+	}
```

Comment:
Thanks.I had to declare ret, and I placed the declaration inside the #ifdef block to avoid potential compiler warnings.

Reply:
Since the polling/interrupt behavior is driver-related, and not really device-related, are you sure it belongs in the device tree? Either way, I agree that there are a few too many compilation options. But I see cases where all three configurations more-or-less make sense...cheers.